telemetry/client.go

```diff
@@ -71,21 +87,28 @@ type PeerCount struct {
 	PeerCount int
 }
 
+type PeerConnFailure struct {
+	FailedPeerId string
+	FailureCount int
+}
+
 type Client struct {
-	serverURL           string
-	httpClient          *http.Client
-	logger              *zap.Logger
-	keyUID              string
-	nodeName            string
-	peerId              string
-	version             string
-	telemetryCh         chan TelemetryRequest
-	telemetryCacheLock  sync.Mutex
-	telemetryCache      []TelemetryRequest
-	telemetryRetryCache []TelemetryRequest
-	nextIdLock          sync.Mutex
-	nextId              int
-	sendPeriod          time.Duration
+	serverURL            string
+	httpClient           *http.Client
+	logger               *zap.Logger
+	keyUID               string
+	nodeName             string
+	peerId               string
+	version              string
+	telemetryCh          chan TelemetryRequest
+	telemetryCacheLock   sync.Mutex
+	telemetryCache       []TelemetryRequest
+	telemetryRetryCache  []TelemetryRequest
+	nextIdLock           sync.Mutex
+	nextId               int
```

Comment:
Just a suggestion, out of this PR
```suggestion
	nextId               atomic.Int32
```